drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1161 @@
+		return -EIO;
+	}
+
+	/* set settings */
+	retval = tmag5273_init_sensor_settings(drv_cfg);
+
+	if (retval < 0) {
+		LOG_ERR("error setting sensor configuration |%d|", retval);
+		return retval;
+	}
+
+	retval = tmag5273_init_device_config(drv_cfg);
+
+	if (retval < 0) {
+		LOG_ERR("error setting device configuration |%d|", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmag5273_driver_api = {
+	.attr_set = tmag5273_attr_set,
+	.attr_get = tmag5273_attr_get,
+	.sample_fetch = tmag5273_sample_fetch,
+	.channel_get = tmag5273_channel_get,
+};
+
+/** Instantiation macro */
+#define TMAG5273_DEFINE(inst)                                                                      \
+	BUILD_ASSERT(IS_ENABLED(CONFIG_CRC) ||                                                     \
+		     (!IS_ENABLED(CONFIG_CRC) && (DT_INST_PROP(inst, crc_enabled) == 0)));         \
+	static const struct tmag5273_config tmag5273_driver_cfg##inst = {                          \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.x_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_DT_AXIS_X),                   \
+		.y_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_DT_AXIS_Y),                   \
+		.z_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_DT_AXIS_Z),                   \
+		.temperature = DT_INST_PROP(inst, temperature),                                    \
+		.meas_range = DT_INST_PROP(inst, range),                                           \
+		.temperature_coefficient = DT_INST_PROP(inst, temperature_coefficient),            \
+		.angle_magnitude_axis = DT_INST_PROP(inst, angle_magnitude_axis),                  \
+		.ch_mag_gain_correction = DT_INST_PROP(inst, ch_mag_gain_correction),              \
+		.operation_mode = DT_INST_PROP(inst, operation_mode),                              \
+		.averaging = DT_INST_PROP(inst, average_mode),                                     \
+		.trigger_conv_via_int = DT_INST_PROP(inst, trigger_conversion_via_int),            \
+		.low_noise_mode = DT_INST_PROP(inst, low_noise),                                   \
+		.ignore_diag_fail = DT_INST_PROP(inst, ignore_diag_fail),                          \
+		IF_ENABLED(CONFIG_CRC, (.crc_enabled = DT_INST_PROP(inst, crc_enabled),))          \
+		.int_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0}),                        \
+		.mask_intb_value = (DT_INST_PROP(inst, trigger_conversion_via_int) &&              \
+						    !DT_INST_PROP(inst, force_mask_int_pin)        \
+					    ? TMAG5273_INT_MASK_INTB_ENABLED                       \
+					    : TMAG5273_INT_MASK_INTB_MASKED),                      \
+		.need_int_pin = (DT_INST_PROP(inst, trigger_conversion_via_int)),                  \
+	};                                                                                         \
+	static struct tmag5273_data tmag5273_driver_data##inst = {0};                              \
```

Comment:
drop the initializer, this is zero initialized automatically

Reply:
You sure? I don't find this with a fast skim through the standard, but afaik `0` default initialization may only be applied for debugging builds based on the compiler.

Reply:
Yes https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/init.c#L172